### DISCUSSION TITLE
How to visualize higher order elemental variable, how transfer them through SolutionUserObjects?

### DISCUSSION THREAD

**Discussion by aaelmeli** [idaholab/moose#23768] [Q&A General] [answered]
Hi

As indicated on MOOSE webpage, [see this link](https://mooseframework.inl.gov/source/auxkernels/RankTwoScalarAux.html) the exodus format does not support higher than `CONSTANT` elemental variable. Now, I want to transfer higher order elemental variable, say `First`) to another simulation using the `SolutionUserObject`, How can I achieve this? only through `XDA` format?

if that is the case, how can I visualize them? currently, I am using Paraview for visualization.

**Comment by GiudGiud** [comment, marked as answer]
Hello

you can visualize with Exodus still, before saving to XDA and after loading from XDA. 

I think we can represent a first order elemental variable fine, just maybe not load them. 
@roystgnr knows more about our exodus output

Guillaume

**Reply by roystgnr** [reply]
IIRC we decimate before writing to Exodus - so you can have a `MONOMIAL` variable of any order, but when you write it to Exodus only the `CONSTANT` component (e.g. the value that would be at the center node of a `QUAD9` or the zeroth vertex of a `TRI`) gets written.

This isn't just an issue with elemental variables; e.g. if we have a `CUBIC` variable on a `QUAD9` it'll be interpolated at those 9 points (and you're down to p=2), and then if you use Paraview IIRC it adds insult to injury by interpolating those 9 points via 8 linear triangles rather than a single biquadratic (at which point you're down to p=1 at h:=h/2).

Exodus (including Nemesis) is our best option for visualization, but make sure you're doing any restarts from XDA/XDR/Checkpoint, and any postprocessing from MOOSE.
